glslc/test/placeholder.py

```diff
@@ -115,3 +116,19 @@ def instantiate_for_glslc_args(self, testcase):
 
     def instantiate_for_expectation(self, testcase):
         return os.path.join(testcase.directory, self.filename)
+
+
+class SpecializedString(PlaceHolder):
+    """Returns a string that has been specialized based on TestCase."""
```

Comment:
This docstring could use a little more detail.  What does specialization mean?  It means template instantiation where $Var is replaced by the Var member of the Testcase.